src/core/lib/transport/handshaker.h

```diff
@@ -122,37 +125,39 @@ class HandshakeManager : public RefCounted<HandshakeManager> {
   /// the arguments.
   void DoHandshake(grpc_endpoint* endpoint, const ChannelArgs& channel_args,
                    Timestamp deadline, grpc_tcp_server_acceptor* acceptor,
-                   grpc_iomgr_cb_func on_handshake_done, void* user_data);
+                   grpc_iomgr_cb_func on_handshake_done, void* user_data)
+      ABSL_LOCKS_EXCLUDED(mu_);
 
  private:
-  bool CallNextHandshakerLocked(grpc_error_handle error);
+  bool CallNextHandshakerLocked(grpc_error_handle error)
+      ABSL_EXCLUSIVE_LOCKS_REQUIRED(mu_);
 
   // A function used as the handshaker-done callback when chaining
   // handshakers together.
-  static void CallNextHandshakerFn(void* arg, grpc_error_handle error);
-
-  // Callback invoked when deadline is exceeded.
-  static void OnTimeoutFn(void* arg, grpc_error_handle error);
+  static void CallNextHandshakerFn(void* arg, grpc_error_handle error)
+      ABSL_LOCKS_EXCLUDED(mu_);
 
   static const size_t HANDSHAKERS_INIT_SIZE = 2;
 
   Mutex mu_;
-  bool is_shutdown_ = false;
+  bool is_shutdown_ ABSL_GUARDED_BY(mu_) = false;
   // An array of handshakers added via grpc_handshake_manager_add().
   absl::InlinedVector<RefCountedPtr<Handshaker>, HANDSHAKERS_INIT_SIZE>
-      handshakers_;
+      handshakers_ ABSL_GUARDED_BY(mu_);
   // The index of the handshaker to invoke next and closure to invoke it.
-  size_t index_ = 0;
-  grpc_closure call_next_handshaker_;
+  size_t index_ ABSL_GUARDED_BY(mu_) = 0;
+  grpc_closure call_next_handshaker_ ABSL_GUARDED_BY(mu_);
   // The acceptor to call the handshakers with.
-  grpc_tcp_server_acceptor* acceptor_;
+  grpc_tcp_server_acceptor* acceptor_ ABSL_GUARDED_BY(mu_);
   // Deadline timer across all handshakers.
```

Comment:
Suggest moving these 3 lines down to line 159, so it's right next to the EE data member.